### ISSUE TITLE
Precision and Recall for each label when testing

### ISSUE THREAD

**Issue by loretoparisi** [facebookresearch/fastText#212]
When testing a model I get an output like

```javascript
{ N: '2191', P: '0.693', R: '0.693' }
```

with Precision `P`, Recall `R` and testset size `N` values.
It could help to get those values for each label in the model. Is that possibile?

**Comment by zhongboyin** [comment]
same question with you, did you get the answer?

**Comment by loretoparisi** [comment]
Not yet. Also there is another issue that I have: the values for P and R are for all my models always the same.

**Comment by zhongboyin** [comment]
also the same to me, while the k =1

**Comment by loretoparisi** [comment]
Yes, confirmed this happens for k=1, no matter test size N, or training set size. Also with different settings for ws, lr, dim, etc I have R and P equals.

**Comment by zhongboyin** [comment]
couldn't more agree

**Comment by AkshitaJha** [comment]
Any updates on this?

**Comment by loretoparisi** [comment]
Any update to have the precision `P` and recall `R` for each label?

**Comment by cpuhrsch** [comment]
Hello @loretoparisi ,

Thank you for your post. It looks like you're using a third party library to use fastText. Unfortunately we can't help you with this. You're probably best advised to follow up with the maintainer of the package, unless you can reproduce an error with the fasttext binary itself (which is part of this repository). Feel free to reopen this issue at any time if this is indeed the case.

Thank you,
Christian

**Comment by Hieunv1996** [comment]
Hi, I'm a fasttext user,
I do not implement code for Precission vs Recall, here my code just show accuracy for each label in training data. Hope it help somebody!
```
def test(model, test_path):
    data = {}
    with open(test_path, encoding='utf8') as fp:
        for line in fp.readlines():
            line = line.strip()
            words = line.split()
            label = words[0]
            pred = model.predict(' '.join(words[1:]))[0][0]
            if label not in data:
                data[label] = {}
                data[label]['correct'] = 0
                data[label]['total'] = 0
            data[label]['total'] += 1
            if label == pred:
                data[label]['correct'] += 1
    summaries = {}
    for key in data:
        summaries[key] = round(data[key]['correct'] / data[key]['total'], 2)
    return summaries, data
```
